daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1097,58 +1097,72 @@ private[lf] object SBuiltin {
   }
 
   // Similar to SBUFetch but is never performed "by key".
-  final case class SBUFetchInterface(ifaceId: TypeConName) extends OnLedgerBuiltin(1) {
+  final case class SBUFetchInterface(ifaceId: TypeConName) extends OnLedgerBuiltin(2) {
     override protected def execute(
         args: util.ArrayList[SValue],
         machine: Machine,
         onLedger: OnLedger,
     ): Unit = {
       val coid = getSContractId(args, 0)
+      val expectedTemplateIdOpt = getSOptional(args, 1).map(typeRep =>
+        typeRep match {
```

Comment:
Maybe worth defining `getTypeRep` or something like that similar to how we have `getSOptional`?

Reply:
Wouldn't help here, since it's a typeRep inside an optional 🙃

Reply:
why not? Can’t you do `getSOptional(args,1).map(getSTypeRep(_))`?

Reply:
What is `getSTypeRep(_)` analogous to?
`getSOptional` and such are applied to args directly.

Also, what is the point in extracting `getSTypeRep` when it's only used here, and still wouldn't be enough to validate the type rep has the format we expect?

Reply:
fair, just seemed like it might be slightly cleaner to write this differently but you’re probably right that it doesn’t matter much.